SDL Event cleanup

yol opened the pull request:
Cleanup of unused SDL event structures

## Description
Remove event structures that were never sent and are handled nowhere.
Remove key/mouse event information that is never used and has undefined meaning in the first place.

## Motivation and Context
When implementing the Wayland windowing system I noticed a lot of events and information that were actually not interpreted anywhere. Some window systems set them, some don't, and the meaning varies. But everything I remove here is not actually used for anything, so I would like to have it removed to make implementing windowing stuff less complicated. 

## How Has This Been Tested?
Searched code for all references and removed them.
Build and run tested on x11.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

pkerling committed:
Get rid of XBMC_MouseButtonEvent.state

pkerling committed:
Get rid of XBMC_KeyboardEvent.state

pkerling committed:
Get rid of XBMC_MouseMotionEvent.[xy]rel

pkerling committed:
Get rid of XBMC_KeyboardEvent.which

pkerling committed:
Get rid of XBMC_MouseMotionEvent.which

pkerling committed:
Get rid of XBMC_MouseButtonEvent.which

pkerling committed:
Get rid of XBMC_RELEASED/XBMC_PRESSED

pkerling committed:
Get rid of XBMC_MouseMotionEvent.state

pkerling committed:
Get rid of XBMC_mouse.h

pkerling committed:
Get rid of XBMC_ActiveEvent

pkerling committed:
Get rid of Xbmc_SysWMEvent

There is no XBMC_syswm.h.

pkerling committed:
Get rid of XBMC_ExposeEvent

Rechi commented:
jenkins build this please

Memphiz commented:
Wow - strange that the key.state is never use. If jenkins is fine on osx it has my ok

pkerling committed:
Get rid of CWinEvents::GetQueueSize

Some platforms implement it, some don't, and it's really never used
outside of the windowing system. So consequently it should be private
to the windowing implementation if it decides to use it.

pkerling commented:
Forgot about one =)

Memphiz commented:
Don't forget to grep in *.m and *.mm files too - the last commit looks odd because it only removes the definition in the header for ios. There must be an implementation of it too somewhere.

pkerling commented:
iOS implementation uses it internally so it's not removed, just made private

garbear commented:
No comment, code is outside my knowledge

pkerling committed:
Get rid of PHANDLE_EVENT_FUNCTION

It is only ever used by Win32, which can as well just use
g_application.OnEvent directly - all others ignore it anyway

pkerling committed:
Get rid of IWinEvents inheriting Observer

Only used by CWinEventsLinux which should inherit Observer directly
instead. It is completely unclear from looking at IWinEvents
what the Notify function is actually meant to receive notifications for.

Rechi commented:
@pkerling this needs a rebase

pkerling commented:
done

koying reviewed (commented):
Nice cleanup :)

MartijnKaijser commented:
Good for merge?

Rechi commented:
With this changes keyboard presses and mouse motion on OS X doesn't do the expected actions.

fritsch commented:
No debuglog no issue

2017-06-10 11:55 GMT+02:00 Rechi <notifications@github.com>:

> With this changes keyboard presses and mouse motion on OS X doesn't do the
> expected actions.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/12272#issuecomment-307555112>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABCfHTYbezs7Ot23_3kzu6JyyFRQbNJPks5sCmgHgaJpZM4N0LkW>
> .
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

pkerling commented:
@Rechi: do you have more details? what does not work? and can you bisect?

FernetMenta commented on xbmc/windowing/osx/WinEventsSDL.cpp:
```diff
@@ -389,17 +377,6 @@ bool CWinEventsSDL::MessagePump()
   return ret;
 }
 
-size_t CWinEventsSDL::GetQueueSize()
```
yre you sure this method can be dropped?

FernetMenta commented:
on osx it stops working with https://github.com/xbmc/xbmc/pull/12272/commits/4f8acfee8b25b7d8bf644af8ae81119174948979

for what reason ever

pkerling commented:
@FernetMenta Ah that makes sense. Good riddance then, doing that is a very bad idea anyway. And thanks for addressing it :)

MilhouseVH commented:
After this PR, OnKey processing in LibreELEC on X86 with X11 is no longer working.

Instead of:
```
23:23:34.366 T:139684275845184   DEBUG: OnKey: down (0xf081) pressed, action is Down
23:23:34.486 T:139684275845184   DEBUG: Keyboard: scancode: 0x74, sym: 0x0112, unicode: 0x0000, modifier: 0x0
23:23:36.798 T:139684275845184   DEBUG: Previous line repeats 1 times.
23:23:36.799 T:139684275845184   DEBUG: OnKey: down (0xf081) pressed, action is Down
23:23:36.921 T:139684275845184   DEBUG: Keyboard: scancode: 0x74, sym: 0x0112, unicode: 0x0000, modifier: 0x0
```

once this PR is included, all that appears in the log when mashing the keyboard is:
```
23:27:40.661 T:139908466358336   DEBUG: ------ Window Init () ------
23:28:20.166 T:139908466358336   DEBUG: Keyboard: scancode: 0x74, sym: 0x0112, unicode: 0x0000, modifier: 0x0
```
There is no `OnKey` processing at all.

Full log: http://sprunge.us/ePTD - at the end of this log I'm alternately pressing the up/down cursor keys and there is no response in the GUI.

Reverting this PR fixes the issue (and #12285 has no effect)

pkerling commented:
Yes #12285 is OS X only.
It's really weird since you see that the keypresses are correctly registered. But I can reproduce it here. Will fix.

pkerling commented:
Same problem as SDL - maps X11 event numbers 1:1 to Kodi events. Interesting that SDL and X11 were identical, too. I'll also check all other windowing implementations to see whether any of them does the same, but it's highly unlikely.

FernetMenta commented:
@pkerling please rebase on my pr or include it in yours

pkerling commented:
I'll do it on top of #12287 
@MilhouseVH please follow that PR for the fix
